modules/llm-cache/storage/file_storage.cc

```diff
@@ -39,78 +41,48 @@ namespace vineyard {
-  int kvStateSize = kvStateList[0].begin()->second.first.length;
+  std::vector<std::string> pathList;
+  std::string dir = rootPath;
+  std::string path;
+  int tokenSize;
+
+  VINEYARD_ASSERT(batchSize > 0);
+  // check if the root path ends with '/'
+  if (dir.back() != '/') {
+    dir += "/";
+  }
 
-  std::vector<std::string> paths;
-  RETURN_ON_ERROR(DefaultGetPathFromPrefix(tokenList, batchSize, paths));
-  for (size_t i = 0; i < tokenList.size() / batchSize; i++) {
+  RETURN_ON_ERROR(hasher->computePathForTokens(tokenList, batchSize,
+                                               splitNumber, pathList));
+  if (pathList.size() == 0) {
+    return Status::OK();
+  }
+  tokenSize = pathList.size() * batchSize;
+  for (size_t i = 0; i < pathList.size(); i++) {
     std::shared_ptr<FileDescriptor> fd;
-
-    // TODO: check if the kv state is already in the cache
-    // TODO: hash conflict
-    RETURN_ON_ERROR(Open(rootPath + paths[i], fd, FileOperationType::WRITE));
-    FileHeader header;
-    size_t fileSize;
-    RETURN_ON_ERROR(GetFileSize(fd, fileSize));
-    LOG(INFO) << "fileSize:" << fileSize;
-    if (fileSize != 0) {
-      Read(fd, &header, sizeof(FileHeader));
-      if (header.layer != layer || header.kvStateSize != kvStateSize) {
-        Close(fd);
-        return Status::Invalid("Layer/KV state size mismatch!");
-      }
-      header.prefixNum += tokenList.size();
-      Seek(fd, 0);
-      Write(fd, &header, sizeof(FileHeader));
-    }
-    {
-      header.prefixNum = tokenList.size();
-      header.layer = layer;
-      header.kvStateSize = kvStateSize;
-      Write(fd, &header, sizeof(FileHeader));
-    }
-    LOG(INFO) << "Header: " << header.prefixNum << " " << header.layer << " "
-              << header.kvStateSize;
-
-    for (size_t j = 0; j < batchSize; j++) {
-      // write
-      int prefixSize = (i * batchSize + j + 1) * sizeof(int);
-      LOG(INFO) << "Write prefix length:" << prefixSize / 4;
-      LOG(INFO) << "write token list:";
-      std::string token_str = "";
-      for (size_t k = 0; k <= j + i * batchSize; k++) {
-        token_str += std::to_string(tokenList[k]) + " ";
-      }
-      LOG(INFO) << token_str;
-      Write(fd, &prefixSize, sizeof(int));
-      Write(fd, tokenList.data(), prefixSize);
-      for (int currentLayer = 0; currentLayer < layer; currentLayer++) {
-        LOG(INFO) << "write:";
-        std::string data;
-        for (int k = 0; k < kvStateSize; k++) {
-          data += std::to_string((
-                      reinterpret_cast<uint8_t*>(kvStateList[i * batchSize + j]
-                                                     .find(currentLayer)
-                                                     ->second.first.data))[k]) +
-                  " ";
+    std::filesystem::path filePath(dir + pathList[i]);
+    RETURN_ON_ERROR(Mkdir(filePath.parent_path().string()));
+    RETURN_ON_ERROR(Open(filePath.string(), fd, FileOperationType::WRITE));
+    RETURN_ON_ERROR(Seek(fd, 0));
```

Comment:
Currently, when open a file, the read/write pointer is at the begin of the file.

Reply:
Make sense to me. Thanks